### PR TITLE
DynamoDB can now use AWS CostExplorer

### PR DESCRIPTION
**Pull request by ryan-j-mci**
Added the ability to use AWS CostExplorer with DynamoDB. This uses the same format as the other plugins, and filters on the tag table-name.

### CODE REVIEW

File: team-cost-reporter/plugins/dynamodb/__init__.py [hunk 1]
```diff
@@ -142,8 +144,33 @@ def getTeamCost(team_name,configMap,debug):
...
+                    log("   getting data from aws using cost explorer")
+                    data = aws_cost_collector.get_costs(days_to_report=days_to_report,
+                                                        granularity=granularity,
+                                                        filter=filter,
+                                                        group_by=group_by,
+                                                        debug=debug)
+                    # We need to convert this to cloudcheckr format
+                    data = cloudcheckr.convert(data, group_by_tag, debug)
+
+    if data and 'aws_dynamodb' in config_plugin:
```
Comment (haggaret) [inline team-cost-reporter/plugins/dynamodb/__init__.py]: Is there a reason why you need the extra 'and' now?

File: team-cost-reporter/plugins/dynamodb/__init__.py [hunk 2]
```diff
@@ -198,4 +225,66 @@ def getTeamCost(team_name,configMap,debug):
                     team_cost['shared'][cc_table_name] = format(float(totalCost),'.2f')
                     totalCost = 0
 
+    elif data and 'aws_ce' in config_plugin:
```
Comment (haggaret) [inline team-cost-reporter/plugins/dynamodb/__init__.py]: Couldn't you just have the one "if data:" and then under that if 'aws_dynamodb' in config_plugin, elif 'aws_ce' in config_plugin...

Or even better, can you recognize the difference in the data somhow?

File: team-cost-reporter/plugins/dynamodb/__init__.py [hunk 3]
```diff
@@ -198,4 +225,66 @@ def getTeamCost(team_name,configMap,debug):
...
+                    if cc_env_name.lower().endswith(suffix.lower()):
+                        if debug: log("Wildcard suffix match found")
+                        match = True
+                elif config_match_env.lower() == cc_env_name.lower():
+                    if debug: log("Exact matching")
+                    match = True
+
+                if match:
+                    # Match found - add cost
+                    if debug: log("*** Matching ecs cluster  found")
```
Comment (haggaret) [inline team-cost-reporter/plugins/dynamodb/__init__.py]: Mentions ECS cluster here...

File: team-cost-reporter/plugins/dynamodb/__init__.py [hunk 4]
```diff
@@ -198,4 +225,66 @@ def getTeamCost(team_name,configMap,debug):
...
+                if len(tag_value) == 2:
+                    cc_env_name = str(tag_value[1].strip().lower())
+                else:
+                    continue
+
+                if debug: log("dynamodb table in report data is %s" % cc_env_name)
+
+                # See if we're dealing with a wildcard or exact match
+                match = False
+                if str(config_match_env).startswith("*") and str(config_match_env).endswith("*"):
```
Comment (haggaret) [inline team-cost-reporter/plugins/dynamodb/__init__.py]: All of this wildcard detection code stuff is common, it's just the string you're looking at is different - could you create a function that takes an input string, does this check and then returns True/False?

**Comment by haggaret** [comment]
Are there any config file changes required? If so, can you update the sample config with an example?

File: team-cost-reporter/plugins/dynamodb/__init__.py (hunk 2 above)
Comment (ryan-j-mci) [reply team-cost-reporter/plugins/dynamodb/__init__.py]: I don't see a way to have this be data dependent, but I have improved the logic.

File: team-cost-reporter/plugins/dynamodb/__init__.py (hunk 4 above)
Comment (ryan-j-mci) [reply team-cost-reporter/plugins/dynamodb/__init__.py]: Done

File: team-cost-reporter/plugins/dynamodb/__init__.py [hunk 5]
```diff
@@ -82,6 +84,36 @@ def _get_table_info(account_creds, region_list, debug=False):
     return account_dynamo_details
 
 
+def _is_match(cc_env_name, config_match_env, debug=False):
```
Comment (haggaret) [inline team-cost-reporter/plugins/dynamodb/__init__.py]: My only comment here is around variable names - could probably be more generic, like 'string_to_check' and ?? No idea...  :)

**Review by haggaret — APPROVED** [review]
Looks good, other than my general comment around variable names :)

**Comment by haggaret** [comment]
Ship it! (Squash your commits and merge to master)

**Pull request merged**